source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -567,61 +620,35 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
-bool ContextImpl::verifySubjectAltName(X509* cert,
-                                       const std::vector<std::string>& subject_alt_names) {
+bool ContextImpl::matchSubjectAltName(
+    X509* cert, const std::vector<Matchers::StringMatcherImpl>& match_subject_alt_name_list) {
   bssl::UniquePtr<GENERAL_NAMES> san_names(
       static_cast<GENERAL_NAMES*>(X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr)));
   if (san_names == nullptr) {
     return false;
   }
   for (const GENERAL_NAME* san : san_names.get()) {
-    switch (san->type) {
-    case GEN_DNS: {
-      ASN1_STRING* str = san->d.dNSName;
-      const char* dns_name = reinterpret_cast<const char*>(ASN1_STRING_data(str));
-      for (auto& config_san : subject_alt_names) {
-        if (dnsNameMatch(config_san, dns_name)) {
-          return true;
-        }
+    for (auto& config_san_matcher : match_subject_alt_name_list) {
+      if (config_san_matcher.match(generalNameAsString(san))) {
```

Comment:
store result of `generalNameAsString(san)` outside of the inner loop.

Reply:
Yeah. I noticed that and fixed it. Next commit will have it